### PR TITLE
Frontend/search songs on spotify

### PR DESCRIPTION
**Pull request by LucaSchmid**
Search Song on spotify for creating  a new post.
Not connected with the backend yet

### CODE REVIEW

File: frontend/src/pages/NewPost/.test.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  })
+
+  it('should have 10 recent played song', async () => {
+    act(() => {
+      render(<NewPost />)
+    })
+
+    await waitFor(() => {
+      const songList = screen.queryAllByRole('listitem')
+      expect(songList).toHaveLength(10)
```
Comment (thomasjakkel) [inline frontend/src/pages/NewPost/.test.tsx:27]: is this always true?

Comment (thomasjakkel) [reply frontend/src/pages/NewPost/.test.tsx:27]: because of mock data?

File: frontend/src/pages/NewPost/component.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        setSongs(songs != null ? songs : [])
+      })
+      .catch((err) => {
+        console.error(err)
+      })
+  }
+
+  const onSongClick = (song: trackObject): void => {
+    setSongs([song])
+    setSongId(song.id ?? '')
```
Comment (thomasjakkel) [inline frontend/src/pages/NewPost/component.tsx:37]: hm, when could the songId be null?
Setting an Id to ' ' and sending an empty id value to the backend seems edgy?

File: frontend/src/pages/NewPost/.test.tsx (hunk 1 above)
Comment (LucaSchmid) [reply frontend/src/pages/NewPost/.test.tsx:27]: exactly, because of mock data

**Pull request merged**